### PR TITLE
Refactor ACP <-> PSP communication for AMD platforms

### PR DESCRIPTION
**Pull request by mkatraga**

### CODE REVIEW

**Comment by sofci** [comment]
Can one of the admins verify this patch?
> reply **test&nbsp;this&nbsp;please** to run this test once

**Comment by vijendarmukunda** [comment]
test this please

**Review by bhiregoudar — DISMISSED** [review]

**Review by sunilkumardommati — DISMISSED** [review]

**Review by vijendarmukunda — DISMISSED** [review]

File: sound/soc/sof/amd/acp.c [hunk 1]
```diff
@@ -217,6 +217,7 @@ int configure_and_run_sha_dma(struct acp_dev_data *adata, void *image_addr,
 			      unsigned int image_length)
 {
 	struct snd_sof_dev *sdev = adata->dev;
+	const struct sof_amd_acp_desc *desc = get_chip_info(sdev->pdata);
```
Comment (plbossart) [inline sound/soc/sof/amd/acp.c:216]: typo in commit title: compleiton

File: sound/soc/sof/amd/acp.c [hunk 2]
```diff
@@ -251,9 +252,11 @@ int configure_and_run_sha_dma(struct acp_dev_data *adata, void *image_addr,
 		return ret;
 	}
 
-	ret = psp_send_cmd(adata, MBOX_ACP_SHA_DMA_COMMAND);
-	if (ret)
-		return ret;
+	if (desc->rev == 3) {
```
Comment (plbossart) [inline sound/soc/sof/amd/acp.c:252]: consider using macros or add a commit to explain magic numbers in code.

**Review by plbossart — CHANGES_REQUESTED** [review]

File: sound/soc/sof/amd/acp.c (hunk 1 above)
Comment (mkatraga) [reply sound/soc/sof/amd/acp.c:216]: will fix it.

File: sound/soc/sof/amd/acp.c (hunk 2 above)
Comment (mkatraga) [reply sound/soc/sof/amd/acp.c:252]: will add comment.

Comment (plbossart) [reply sound/soc/sof/amd/acp.c:252]: it's renoir, not renior....

File: sound/soc/sof/amd/acp.h [hunk 3]
```diff
@@ -61,12 +61,12 @@
 #define HOST_BRIDGE_CZN				0x1630
 #define HOST_BRIDGE_RMB				0x14B5
 #define ACP_SHA_STAT				0x8000
-#define ACP_PSP_TIMEOUT_COUNTER			5
+#define ACP_PSP_TIMEOUT				1000000
```
Comment (plbossart) [inline sound/soc/sof/amd/acp.h]: units? US, NS?

Comment (mkatraga) [reply sound/soc/sof/amd/acp.h]: that's US, its a miss will updates as  ACP_PSP_TIMEOUT_US.

**Review by plbossart — APPROVED** [review]

**Comment by plbossart** [comment]
test this please

**Review by vijendarmukunda — APPROVED** [review]

**Review by sunilkumardommati — APPROVED** [review]

**Review by bhiregoudar — APPROVED** [review]

**Comment by mkatraga** [comment]
Hi,

@plbossart Can you please help to merge these patches.

**Pull request merged**